import/import-url/update: --no-download improvements (#3930)
iterative/dvc.org: content/docs/command-reference/import.md

  > [Importing and updating fixed revisions](#example-importing-and-updating-fixed-revisions)
  > example below).

- `--no-exec` - create the import `.dvc` file without accessing `url` (assumes
  that the data source is valid). This is useful if you need to define the
  project imports quickly, and import the data later (use `dvc update` to finish
  the operation(s)).

- `--no-download` - create the import `.dvc` file including the source data

>   [version](https://git-scm.com/docs/revisions) but without downloading the

Reviewer: the import .dvc file saves both the repo URL and the revision (Git commit) so maybe we should keep this more general to "source data information" like in https://github.com/iterative/dvc/pull/8341#pullrequestreview-1122585535.

  associated data. This is useful if you need track changes in remote data
  without using local storage space (yet). The data can be downloaded later

>   using `dvc pull`. File hashes can be updated using `dvc update --no-download`.

Reviewer: What file hashes? :p this is `dvc import`

  > [Importing and updating fixed revisions](#example-importing-and-updating-fixed-revisions)
  > example below).

- `--no-exec` - create the import `.dvc` file without accessing `url` (assumes
  that the data source is valid). This is useful if you need to define the
  project imports quickly, and import the data later (use `dvc update` to finish
  the operation(s)).

- `--no-download` - create the import `.dvc` file including
  [hash values](/doc/user-guide/project-structure/dvc-files#dependency-entries)

>   for the remote file but without downloading the associated data. This is

Reviewer: Wait. There are no hash values in `deps` for `import` .dvc files, are there? It tracks a `repo` + `rev_lock` last time I checked.

Author: Good catch! You're right, there's no hash values in deps for `import --no-download`, which means that the dependencies aren't actually locked. Will need to fix this in `dvc` too. c.c. @dberenbaum

Reviewer 2: Yup, great catch @jorgeorpinel!

Author: Fixed with https://github.com/iterative/dvc/pull/8341. Will push an update here soon.

> - `--no-download` - create the import `.dvc` file including the
>   [version](https://git-scm.com/docs/revisions) for the external dependency but
>   without downloading the associated data. This is useful if you need track

Reviewer: Term "external dependency" is out of context in this page. The description has this phrase though: "[.dvc](https://dvc.org/doc/user-guide/project-structure/dvc-files) files support references to data in an external DVC repo..." so maybe something like:

```suggestion
- `--no-download` - create the import `.dvc` file including the
  reference to the external data source but
  without downloading the associated data. This is useful if you need track
```

Or if we want to keep the link to Git revision concept (unclear it's helpful here IMO though), maybe something like

```suggestion
- `--no-download` - create the import `.dvc` file including the source data
  [version](https://git-scm.com/docs/revisions) but
  without downloading the associated data. This is useful if you need track
```

  you need track changes in remote data without using local storage space (yet).

>   the data can be downloaded later using `dvc pull`, but this will fail if the

Reviewer: ```suggestion
  The data can be downloaded later using `dvc pull`, but this will fail if the
```

>   longer matches the hash values. File hashes can be updated using

Reviewer 2: Is it possible that `dvc pull` on an `import` will fail? Since it relies on a Git hash, seems like this shouldn't be an issue for `import`.

Author: Good point. Well, the git hash might be missing, but I think we can safely remove "but this will fail..."

Author: Removed the phrase. Note that if the hash does exist, dvc will print:
> ERROR: failed to pull data from the cloud - unknown Git revision 'badhash'

  > [Importing and updating fixed revisions](#example-importing-and-updating-fixed-revisions)
  > example below).

- `--no-exec` - create the import `.dvc` file without accessing `url` (assumes
  that the data source is valid). This is useful if you need to define the
  project imports quickly, and import the data later (use `dvc update` to finish
  the operation(s)).

- `--no-download` - create the import `.dvc` file including

>   [hash values](https://dvc.org/doc/user-guide/project-structure/dvc-files#output-entries)

Reviewer: ```suggestion
  [hash values](/doc/user-guide/project-structure/dvc-files#output-entries)
```

  the remote file but without downloading the associated data. This is useful if

>   you need track changes in remote data but do not (yet) need to download data
>   to the local workspace. Data can be later downloaded using `dvc pull`, but
>   will fail if the data matching the file's hash values is no longer available.
>   When remote data changes, file hashes in the `.dvc` file can be updated using
>   `dvc update --no-download`.

Reviewer: ```suggestion
  you need track changes in remote data without using local storage space (yet).
  the data can be downloaded later using `dvc pull`, but this
  will fail if the `url` no longer matches the hash values.
  File hashes can be updated using
  `dvc update --no-download`.
```
